config/main.py

```diff
@@ -1777,7 +1778,15 @@ def vrf_add_management_vrf(config_db):
     config_db.mod_entry('MGMT_VRF_CONFIG', "vrf_global", {"mgmtVrfEnabled": "true"})
     mvrf_restart_services()
 
-def vrf_delete_management_vrf(config_db):
+    cmd = "cat /proc/net/route | grep -E \"eth0\s+00000000\s+[0-9A-Z]+\s+[0-9]+\s+[0-9]+\s+[0-9]+\s+202\" | wc -l"
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    output = proc.communicate()
+    #print "output: {}".format(int(output[0]))
+    if int(output[0]) >= 1:
+        cmd="ip -4 route del default dev eth0 metric 202"
+        os.system(cmd)
```

Comment:
>system [](start = 11, length = 6)

Better to use `Popen.communicate`, and check `Popen.returncode`

Reply:
Resolved, submitted patch

Reply:
Thanks! Instead of checking `err`, better to check `Popen.returncode`

---
In reply to: [536376077](https://github.com/Azure/sonic-utilities/pull/1289#discussion_r536376077) [](ancestors = 536376077)